### ISSUE TITLE
Small Improvements for Docs Page

### ISSUE THREAD

**Issue by patricoferris** [ocaml/v2.ocaml.org#1236]
This would be a good first issue for anyone looking to get started improving and contributing to OCaml.org. 

### IFrame Titles 

Using [`pa11y`](https://github.com/pa11y/pa11y) to perform an accessibility audit on the https://ocaml.org/docs page using the [WCAG2AA](https://www.w3.org/TR/WCAG21/) standard revealed a some small improvement that can be made: 

 - The videos (which are `iframe` elements) should contain a descriptive `title` attribute -- you can read more about this [from Google's lighthouse automation tool](https://web.dev/frame-title/?utm_source=lighthouse&utm_medium=devtools). Note that titles will also need to be added to the `fr` page.

### Mobile Layout

Whilst using the page on mobile I noticed the buttons linking to the OCaml manual appear a bit oddly. Currently they overlap with the title of the following section leading to a somewhat confusing layout. This image should convey this: 

<img width="335" alt="Screenshot 2021-02-09 at 11 25 02" src="https://user-images.githubusercontent.com/20166594/107357019-7511be80-6ac9-11eb-8a7a-1d0ddafa9a72.png">

It would be nice if this was cleaner and well-separated on mobile, but retained its layout for desktop which I think works nicely.

**Comment by nibble0101** [comment]
@avsm  Can I work on this? Pretty much my first time here.

**Comment by avsm** [comment]
@nibble0101 please do! Just ping on here if you run into any blockers trying to fix it.

**Comment by Shubham-Vishwakarma** [comment]
@nibble0101 Are you working on this?
If not please let me know
I would like to give it a try

**Comment by nibble0101** [comment]
@Shubham-Vishwakarma  I am working on it.

**Comment by nibble0101** [comment]
@avsm  I have managed to clone this repository and set up the project on my computer. Is there a script I can run to generate the html file to display in the browser instead of the markdown preview?

**Comment by patricoferris** [comment]
Hi @nibble0101,

Just jumping in to try and unblock you quickly. The site uses a Makefile to run the necessary scripts to generate the HTML version of the site. This actually might be a great opportunity for you to try out the WIP [contributing document](https://github.com/ocaml/ocaml.org/pull/1241) that I am drafting and provide feedback on it? 

The summary of it is you need to: 
 - make sure you have `opam` installed (links in that document) and `make`
 - run `make deps` to get the dependencies for this project 
 - run `make` -- this will take a bit of time, but when it is finished you should see an `ocaml.org` directory which contains the site, you can then open this file (the `index.html` file) in your browser and navigate your modified version of the site. 
 
Your thoughts and opinions on the CONTRIBUTING.md document are more than welcome :))

**Comment by nibble0101** [comment]
Hi @patricoferris,
Thanks for the quick response. It has been very helpful. I was able to run `make` and generate the `ocaml.org` directory. I actually did run `make` before posting the above message asking for help but I cancelled the process because I was impatient and wasn't  sure whether I was doing the right thing. About the contributing document, I will definitely check it out and give you my feedback though the summary you provided above worked for me. Thanks

**Comment by nibble0101** [comment]
@avsm  and @patricoferris  I have a couple of observations here.
1. I faced some challenge working with `.md` files because they are not properly formatted. Wouldn't it be easier to configure a tool to do formatting before making a commit so that each PR follows a consistent style? I changed a few lines of code but because I formatted the code in the file I am working on for readability, the linked PR is showing I changed quite a number of lines of code.
2. Running the page through [HTML validation service](https://validator.w3.org) emits quite a number of errors and warnings for the documentation page. I get the following errors and warnings. Some are quite easy to fix but others require a bit of work if the current styling is to be maintained.
- Error: The element `input` must not appear as a descendant of the `a` element.
- Error: Attribute `webkitallowfullscreen` not allowed on element `iframe` at this point
- Error: Attribute `mozallowfullscreen` not allowed on element `iframe` at this point.
- error: The `frameborder` attribute on the iframe element is obsolete. Use CSS instead
- Warning: The `type` attribute is unnecessary for JavaScript resources.
- Warning: Consider using the `h1` element as a top-level heading only (all `h1` elements are treated as top-level headings by many screen readers and other tools).
- Warning: The `language` attribute on the script element is obsolete. You can safely omit it.
 **To get the above errors and warnings, copy and paste the URL of the documentation page in  [this](https://validator.w3.org) HTML validator**.
Let me know if you want them fixed as well.

**Comment by patricoferris** [comment]
Awesome @nibble0101!

Hopefully we can merge your PR into the staging branch to double-check everything looks good soon. 

Regarding your other points: 
 1. Re:formatting -- on the surface I think this sounds like a good idea, but I can't speak for everyone involved. I think opening a new issue would be a good idea. I think the main talking point will be "which tool to configure". I'm not sure OCaml has such a tool (probably some combination of [omd](https://github.com/ocaml/omd) and [tyxml](https://github.com/ocsigen/tyxml) could do it...) so it would probably mean depending on an entire other ecosystem e.g. npm+node. 
 2. The errors from the valid HTML check should be fixed. Feel free to open a separate PR for them, or describe them in an issue too for others to have a go with. It sounds like a good candidate for a `good-first-issue` :))

**Comment by nibble0101** [comment]
Thanks @patricoferris 
I will try fixing them and then make a PR in the coming days

**Comment by nibble0101** [comment]
@patricoferris  I have been fixing the invalid HTML errors but I have failed to figure out which script is responsible for converting specific markdown file I am working on to HTML. I need to remove `type` attribute from `script` tags which I believe  are generated at the time of converting markdown file to HTML.

**Comment by patricoferris** [comment]
Haven't completely dived that deeply into it, but I would assume it is pulled in from the main template for HTML pages e.g. https://github.com/ocaml/ocaml.org/blob/master/template/main.mpp#L117

**Comment by nibble0101** [comment]
@patricoferris  After making  some changes to `main.mpp` file, running `make` hangs at a certain point as shown in the screenshot below. I am working from a branch other than master. I would have expected some kind of error if my change broke something but it gets stuck at the point of building one of the files I changed. I need some guidance here.
![image](https://user-images.githubusercontent.com/52580190/109802115-7483c800-7c30-11eb-95e4-cedbc4e63e11.png)

**Comment by patricoferris** [comment]
@nibble0101 please can you open a separate PR (label it work in progress if you want) with these changes, that will make it a lot easier for me to help as I'll be able to see the changes and run the code locally. I think I've found the branch, is it https://github.com/nibble0101/ocaml.org/tree/fix-html-validation-errors-on-documentation-page ?

Just as quick point, if you could undo and disable the reformatting of files this would be very much appreciated as for fixes that are not about formatting it introduces a lot of noise and makes review/guidance a little more time-consuming -- thanks.

**Comment by nibble0101** [comment]
@patricoferris  Sure. I will revert the changes and make a PR.
